### PR TITLE
CLN: Follow-ups to #24024

### PR DESCRIPTION
**Pull request by jbrockmendel**
grepped for TODO in the affected files and took care of the easy ones.

Addresses a flake8 complaint in MultiIndex (no idea why it is only showing up locally)

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -681,6 +673,10 @@ def searchsorted(self, value, side='left', sorter=None):
         if isinstance(value, compat.string_types):
             value = self._scalar_from_string(value)
 
+        if isinstance(value, ABCIndexClass):
+            # unbox e.g. DatetimeIndex -> DatetimeArray
+            value = value._values
```
Comment (jbrockmendel) [inline pandas/core/arrays/datetimelike.py]: these lines got added for an unsuccessful attempt to make DatetimeIndex.searchsorted pass through to _data.  long-term i think that is still the right thing to do, but until then i need to revert these lines

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3082,7 +3082,7 @@ def _box_item_values(self, key, values):
     def _maybe_cache_changed(self, item, value):
         """The object has called back to us saying maybe it has changed.
         """
-        self._data.set(item, value, check=False)
```
Comment (jbrockmendel) [inline pandas/core/generic.py]: check is never True, removed the corresponding code in internals.managers and internals.blocks

File: pandas/core/indexes/datetimelike.py [hunk 3]
```diff
@@ -31,23 +31,24 @@
...
     """
+
     def method(self, *args, **kwargs):
-        return getattr(self._eadata, name)(*args, **kwargs)
+        return array_method(self._data, *args, **kwargs)
 
-    method.__name__ = name
-    # TODO: docstrings
+    method.__name__ = array_method.__name__
+    method.__doc__ = array_method.__doc__
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimelike.py]: this should address @jorisvandenbossche's point that docstrings are lost in the existing implementation

**Comment by jreback** [comment]
lgtm.

**Comment by jbrockmendel** [comment]
great.  after this will be a missing-test pass

**Pull request merged**